services/port_manager/src/main/java/com/futurewei/alcor/portmanager/request/RequestManager.java

```diff
@@ -58,16 +64,28 @@ public void sendRequest(IRestRequest request) throws Exception {
     }
 
     public void sendRequestAsync(IRestRequest request, CallbackFunction callback) {
-        CompletableFuture future = CompletableFuture.supplyAsync(() -> {
-            try {
-                sendRequest(request, callback);
-            } catch (Exception e) {
-                throw new CompletionException(e);
-            }
-            return null;
-        }, AsyncExecutor.executor);
+        Span pSpan = tracer.activeSpan();
+        Span span;
+        if (pSpan != null) {
+            span = tracer.buildSpan("alcor-port-async").asChildOf(pSpan.context()).start();
+        } else {
+            span = tracer.buildSpan("alcor-port-async").start();
+        }
 
-        addFuture(request, future);
+        try (Scope scope = tracer.scopeManager().activate(span)) {
+            CompletableFuture future = CompletableFuture.supplyAsync(() -> {
+                try (Scope cscope = tracer.scopeManager().activate(span)) {
+                    try {
+                        sendRequest(request, callback);
+                    } catch (Exception e) {
+                        throw new CompletionException(e);
+                    }
+                    return null;
+                }
+            }, AsyncExecutor.executor).thenRun(span::finish);
+
+            addFuture(request, future);
+        }
```

Comment:
Do we need catch for the external try block?

Reply:
Yes. We can remove the outer try-catch.